### PR TITLE
Prometheus: Metric encyclopedia, return results in table format

### LINKED ISSUE #64705: Metric Encyclopedia: Show results in table, no more expanding. Select button on right inline row.
The Metric Encyclopedia Modal currently displays results in a list of cards that expand showing the metadata and 'use this metric' button. We would like to display the results in a table to always display the meta data of type and description and show the 'use this metric' button in line with the metric name.

The table structure can look like the following. 
```
| Metric Name | Type    | Description            | Action          | 
|-------------|---------|------------------------|-----------------|
| Alert       | Counter | Counts the alert event | Use this metric | 
| temperature | Gauge   | Measure of the temp    | Use this metric | 
|             |         |                        |                 | 
```

We are giving the modal a greater width so that the table columns can be more visible. For "Description" column text that is very long, we can consider giving the overlap a vertical scroll.

### PR DESCRIPTION
**Pull request by bohandley**
Fixes https://github.com/grafana/grafana/issues/64705

What is this change? After collaborating with UX we are now returning metric encyclopedia results in a table format.
Screenshot 2023-03-15 at 3 05 37 PM

Why do we need it? 
- we have more room now due to widening the modal
- hidden type and description in the previous cards was confusing
- hiding the button added an extra click to select a metric
- now we have the same amount of click to add a metric as with the MetricSelect component.
  - click the button
  - focus in text, type if you want
  - select metric with one click

### CODE REVIEW

File: public/app/plugins/datasource/prometheus/querybuilder/components/MetricEncyclopediaModal.tsx [hunk 1]
```diff
@@ -410,6 +410,49 @@ export const MetricEncyclopediaModal = (props: Props) => {
...
+            fuzzySearchQuery: fuzzySearchQuery,
+            fullMetaSearch: fullMetaSearch,
+            selectedTypes: selectedTypes,
+            letterSearch: letterSearch,
+          });
+          onClose();
+        }}
+      >
+        Use this metric
+      </Button>
```
Comment (gwdawson) [inline public/app/plugins/datasource/prometheus/querybuilder/components/MetricEncyclopediaModal.tsx:439]: for this i think having a column of solid blue buttons is too harsh. i would personally prefer one of the following:
```
<Button variant=“primary” fill=“outline” ...rest />
<Button variant=“secondary” fill=“solid” ...rest />
```

feel free to check with UX if you are not sure.

**Comment by gtk-grafana** [comment]
Yeah I love this, and this makes every part of this UI like way better. 

Only a few notes this time:
Rows without descriptions: the view is all whacked for smaller screens:
image

You also get the above view when the metadata is still loading, which can take a while for some users, not sure if it's worth adding a loading state or not, especially if we fix the styles when columns are empty.

Really great work on this, huge improvement!! I'm so excited for y'all!

File: public/app/plugins/datasource/prometheus/querybuilder/components/MetricEncyclopediaModal.tsx [hunk 2]
```diff
@@ -560,67 +602,7 @@ export const MetricEncyclopediaModal = (props: Props) => {
...
-                isOpen={openTabs.includes(metric.value)}
-                collapsible={true}
-                onToggle={() =>
-                  setOpenTabs((tabs) =>
-                    // close tab if it's already open, otherwise open it
-                    tabs.includes(metric.value) ? tabs.filter((t) => t !== metric.value) : [...tabs, metric.value]
-                  )
-                }
-              >
-                <div className={styles.cardsContainer}>
```
Comment (gwdawson) [inline public/app/plugins/datasource/prometheus/querybuilder/components/MetricEncyclopediaModal.tsx:581]: `cardsContainer`, `card` and `metadata` are all styles that are no longer used. these can be deleted from `getStyles()`

**Review by itsgareth — COMMENTED** [review]
left a few suggestions, but its looking good!

**Comment by bohandley** [comment]
> Yeah I love this, and this makes every part of this UI like way better.
> 
> Only a few notes this time: Rows without descriptions: the view is all whacked for smaller screens: image
> 
> You also get the above view when the metadata is still loading, which can take a while for some users, not sure if it's worth adding a loading state or not, especially if we fix the styles when columns are empty.
> 
> Really great work on this, huge improvement!! I'm so excited for y'all!

Definitely need fixes for smaller screens and the loading state has been moved around a few times. I was thinking about putting it by the search text input. Rows without description are metrics without metadata so we lose a lot of space to that, but I think most metrics will have type and description. Most of the metrics without metadata are coming from that dev prom instance.

File: public/app/plugins/datasource/prometheus/querybuilder/components/MetricEncyclopediaModal.tsx (hunk 2 above)
Comment (bohandley) [reply public/app/plugins/datasource/prometheus/querybuilder/components/MetricEncyclopediaModal.tsx:581]: Added!

File: public/app/plugins/datasource/prometheus/querybuilder/components/MetricEncyclopediaModal.tsx (hunk 1 above)
Comment (bohandley) [reply public/app/plugins/datasource/prometheus/querybuilder/components/MetricEncyclopediaModal.tsx]: Changed to secondary!

**Review by itsgareth — APPROVED** [review]
LGTM!
unless @gtk-grafana has any suggestions i'm happy for this to be merged

**Comment by gtk-grafana** [comment]
Looks good!

**Pull request merged**